filelog.go

```diff
@@ -87,6 +91,50 @@ func (w *FileLogWriter) Close() {
 	<- w.completed
 }
 
+// Track write failures and prints to stderr when possible. If err is nil, we'll try to clear the failures
+func (w *FileLogWriter) handleWriteFailure(err error) {
+	// Try to note any previous failures
+	if w.writeFailures != 0 {
+		_, fprintfErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Dropped %d previous log messages\n", w.filename, w.writeFailures)
+		if fprintfErr != nil {
+			// If we can't print now, exit early and try later
+			if err != nil { w.writeFailures += 1 }
+			return
+		} else {
+			w.writeFailures = 0
+		}
+	}
+	// If we have a current failure, attempt to print it
+	if err != nil {
+		_, fprintfErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Write failed: %v\n", w.filename, err)
+		if fprintfErr != nil {
+			w.writeFailures += 1
+		}
```

Comment:
The following sequence was what I was thinking:
1. Some non-zero number of write failures have occurred when we enter handleWriteFailure()
2. Because it is non-zero we attempt to print `"FileLogWriter(%q): Dropped %d previous log messages\n"`.  However this fails. The number of write failures is incremented.
3. The error is non-nil so we attempt to print it, and we succeed.  The number of write failures is not incremented, but it is not reset. We exit the function.
4. We enter the function again.  At this point we still have a non-zero number of write failures. So we attempt to print `"FileLogWriter(%q): Dropped %d previous log messages\n"`, and we succeed.

Reading through this I'm seeing that we wouldn't get two notices, as the first would always reset the count.  But I think I was confused because there is a case where we say we dropped writes after a successful write not before it, which is mildly confusing.  But seeing all that has to happen to hit that case then I think it would be silly to worry about it.  I think this is good as is.